### ISSUE TITLE
Duplicated ACME challenge for cloudflare provider

### ISSUE THREAD

**Issue by MiniDigger** [containous/traefik#3887]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

* Configured acme dns challenge with the cloudflare provider
* Configured domains to use wildcard certs

### What did you expect to see?

* traefik should create txt records for my cloudflare domains and then issue LE certs

### What did you see instead?

* traefik creates a txt record, deletes it, creates a new one and deletes that too (after the challenge failed)
* dns challenge expects second txt record but gets first, thus fails

### Output of `traefik version`: (_What version of Traefik are you using?_)

1.6.6, build on 2018-08-20

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

running in docker 17.12.1-ce on ubuntu, cloudflare provider. 
acme config:  
```
[acme]
#caServer = "https://acme-staging.api.letsencrypt.org/directory"
email = "----------"
storage = "/acme.json"
entryPoint = "https"
acmeLogging = true
onDemand = true
onHostRule = true
  [acme.dnsChallenge]
  provider = "cloudflare"
  delayBeforeCheck = 0
  [acme.httpChallenge]
  entryPoint = "http"

[[acme.domains]]
  main = "*.-------"
  sans = ["----"]
<more domains>
```
arguments:  
```
CMD ["--api","--docker","--docker.domain=docker.localhost","--logLevel=INFO"]
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

I currently have no way to change to log level (I am doing this all via my phone on vacation...)
```
time="2018-09-14T07:20:29Z" level=info msg="Renewing certificate from LE : {Main:*.------ SANs:[------]}"
time="2018-09-14T07:20:59Z" level=error msg="Error renewing certificate from LE: {*.------ [------]}, acme: Error -> One or more domains had a problem:\n[------] acme: Error 403 - urn:ietf:params:acme:error:unauthorized - Incorrect TXT record \"<content of first txt record>\" found at _acme-challenge.------\n"
```

cloudflare audit log:
(first added txt is the one that gets return, second one is what acme expects)
```
2018-09-14T07:20:59Z | delete | true | account | {"actor_email":"------","content":"------","name":"_acme-challenge.------","zone_id":"------","zone_name":"------"}
2018-09-14T07:20:46Z | add | true | DNS_record | {"actor_email":"------","content":"------","name":"_acme-challenge.------","proxied":false,"ttl":"120","type":"TXT","zone_id":"------","zone_name":"------"}
2018-09-14T07:20:45Z | delete | true | account | {"actor_email":"------","content":"------","name":"_acme-challenge.------","zone_id":"------","zone_name":"------"}
2018-09-14T07:20:31Z | add | true | DNS_record | {"actor_email":"------","content":"------","name":"_acme-challenge.------","proxied":false,"ttl":"120","type":"TXT","zone_id":"------","zone_name":"---------"}
```

**Comment by nmengin** [comment]
Hello @MiniDigger ,

Many thanks for your interest in the project.

The bug you mentionned comes from the TTL set when DNS Challenges are solved for both wildcard domain and its basename.

It is solved in the last versionsof Træfik (1.7RC4) as you can see in the PR #3675 and in this closed issue #3761.

Can you try to use the last version of Træfik and give us your feedback?

Thanks in advance.

**Comment by cowai** [comment]
I am running 1.7.5 and I think I have the same issue.

´´´
Error renewing certificate from LE: acme: Error -> One or more domains had a problem:\n[subsub.subdomain.domain.no] acme: Error 403 - urn:ietf:params:acme:error:unauthorized - No TXT record found at _acme-challenge.subsub.subdomain.domain.no\n

´´´
On cloudflare I see two _acme-challenge entries for each of my domains.

I tried deleting them from cloudflare and restarting my traefik cluster but they just came back with same error and duplicate entries in cloudflare.
